File: algorithms/cpp/distributeCandies/distributeCandies.cpp
// Source: https://leetcode.com/problems/distribute-candies/
// Author: Miao Zhang
// Date:   2021-02-23

class Solution {
public:
    int distributeCandies(vector<int>& candyType) {
        unordered_set<int> kinds(candyType.begin(), candyType.end());
        return min(kinds.size(), candyType.size() / 2);
    }
};

File: algorithms/python/distributeCandies/distributeCandies.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/distribute-candies/
# Author: Miao Zhang
# Date:   2021-02-23

class Solution:
    def distributeCandies(self, candyType: List[int]) -> int:
        return min(len(set(candyType)), len(candyType) // 2)
